src/main/java/org/gridsuite/study/server/service/SensitivityAnalysisService.java

```diff
@@ -129,6 +131,27 @@ public String getSensitivityAnalysisResult(UUID nodeUuid, String selector) {
         return result;
     }
 
+    public byte[] exportSensitivityResultsAsCsv(UUID nodeUuid, SensitivityAnalysisCsvFileInfos sensitivityAnalysisCsvFileInfos) {
+        Optional<UUID> resultUuidOpt = networkModificationTreeService.getComputationResultUuid(nodeUuid, ComputationType.SENSITIVITY_ANALYSIS);
+        if (resultUuidOpt.isEmpty()) {
+            return null;
+        }
+
+        // initializing from uri string (not from path string) allows build() to escape selector content
+        URI uri = UriComponentsBuilder.fromUriString(sensitivityAnalysisServerBaseUri)
+                .pathSegment(SENSITIVITY_ANALYSIS_API_VERSION, RESULTS, resultUuidOpt.get().toString(), "csv")
+                .build()
+                .encode()
+                .toUri();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+
+        HttpEntity<SensitivityAnalysisCsvFileInfos> httpEntity = new HttpEntity<>(sensitivityAnalysisCsvFileInfos, headers);
+        return restTemplate.exchange(uri, HttpMethod.POST, httpEntity, byte[].class).getBody();
+
```

Comment:
To propagate exception as other methods

try {
           return restTemplate.exchange(uri, HttpMethod.POST, httpEntity, byte[].class).getBody();
} catch (HttpStatusCodeException e) {
           if (HttpStatus.NOT_FOUND.equals(e.getStatusCode())) {
                throw new StudyException(SENSITIVITY_ANALYSIS_NOT_FOUND);
            } else {
                throw handleHttpError(e, SENSITIVITY_ANALYSIS_ERROR);
            }
        }

Reply:
Done